### DISCUSSION TITLE
Request: Call custom endpoint from admin App

### DISCUSSION THREAD

**Discussion by C14r** [directus/directus#2776] [Archive]
Hello,

i would love to have an option to call a custom endpoint from the admin app. In my case I have a custom endpoint for generating a PDF. Now while I'm viewing the item, or in batch mode, i want to call the endpoint and generate (download) the pdf.

It's a bite like Nova Actions (https://nova.laravel.com/docs/3.0/actions/defining-actions.html).

**Comment by benhaynes** [comment]
In the upcoming v9, we allow for custom page sidebar components (on the right-side of the app). This could be used to create a button to hit your custom API endpoint. I also think it would be interesting to allow for custom header action buttons, although I still see that space as reserved for system buttons.

The question would be, do we need to allow for this without custom code (UI only), which always makes things a bit more opinionated and complex.

**Comment by C14r** [comment]
Thanks for your quick response. 

Thats cool, i understand that the header space is reservt for system buttons. But probably one drop down button for custom actions would not steel much space.

It's would be great to have not too write custom code. But for me it's OK with the custom code. Without custom code is hard to set up.

**Reply by benhaynes** [reply]
Agree... we'll move this to a Feature Request and see if it gets traction.
